### PR TITLE
fix(tests): Fix flaky unfurl test

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
I'm suspecting that these tests are trampling on each other as they all shared the same project id. This could mean that the cleanup of one test wipes the eventstore of another test or multiple setup results in more events than expected.

I'm dealing with this case by having each test have their own unique project.

### CODE REVIEW

File: tests/sentry/integrations/slack/test_unfurl.py [hunk 1]
```diff
@@ -188,14 +191,15 @@ def test_unfurl_discover_previous_period(self, mock_generate_chart):
...
         self.store_event(
-            data={"fingerprint": ["group2"], "timestamp": min_ago}, project_id=self.project.id
+            data={"fingerprint": ["group2"], "timestamp": min_ago}, project_id=unique_proj.id
         )
         self.store_event(
-            data={"fingerprint": ["group2"], "timestamp": min_ago}, project_id=self.project.id
+            data={"fingerprint": ["group2"], "timestamp": min_ago}, project_id=unique_proj.id
         )
 
-        url = f"https://sentry.io/organizations/{self.organization.slug}/discover/results/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&project={self.project.id}&query=&sort=-timestamp&statsPeriod=24h&yAxis=count_unique%28user%29&yAxis=count%28%29"
```
Comment (mgaeta) [inline tests/sentry/integrations/slack/test_unfurl.py:198]: These giant URLs are going to be hard to maintain. Can you rewrite them with `urlencode()`?

File: tests/sentry/integrations/slack/test_unfurl.py [hunk 2]
```diff
@@ -60,8 +60,9 @@ def setUp(self):
 
     def test_unfurl_issues(self):
         min_ago = iso_format(before_now(minutes=1))
+        unique_proj = self.create_project()
```
Comment (mgaeta) [inline tests/sentry/integrations/slack/test_unfurl.py:63]: You could probably get away with doing:
```py
def setUp() -> None:
  super().setUp()
  self.project = self.create_project()
```
instead.

File: tests/sentry/integrations/slack/test_unfurl.py (hunk 1 above)
Comment (AniketDas-Tekky) [reply tests/sentry/integrations/slack/test_unfurl.py:198]: I don't think we can this case. There are some repeated parameters: `?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&`

File: tests/sentry/integrations/slack/test_unfurl.py (hunk 2 above)
Comment (AniketDas-Tekky) [reply tests/sentry/integrations/slack/test_unfurl.py]: Yeah, this should probably work. Only reason I see it not working is if the multiple test runners share the same test class object.... but that seems really unlikely

File: tests/sentry/integrations/slack/test_unfurl.py (hunk 1 above)
Comment (mgaeta) [reply tests/sentry/integrations/slack/test_unfurl.py]:
```py
params = urlencode([
    ("field", "event.type"),
    ("field", "project"),
    ("field", "timestamp"),
    ("field", "title"),
    ("field", "user.display"),
    ("name", "All+Events"),
    ("project", self.project.id),
    ("query", ""),
    ("sort", "-timestamp"),
    ("statsPeriod", "24h"),
    ("yAxis", "count_unique(user)"),
    ("yAxis", "count()"),
])
url = f"{BASE_URL}/organizations/{self.organization.slug}/discover/results/?{params}"
```
Its a little painful but it kinda works

Comment (mgaeta) [reply tests/sentry/integrations/slack/test_unfurl.py]: Oops I see the change is gone, nvm.

File: tests/sentry/integrations/slack/test_unfurl.py [hunk 3]
```diff
@@ -54,6 +54,7 @@ def test_match_link(url, expected):
 class UnfurlTest(TestCase):
     def setUp(self):
         super().setUp()
+        self.project = self.create_project()
```
Comment (mgaeta) [inline tests/sentry/integrations/slack/test_unfurl.py:57]: I'd leave a comment here explaining why we're redefining project.

**Review by mgaeta — APPROVED** [review]

**Pull request merged**